subprojects/core/src/integTest/groovy/org/gradle/execution/taskpath/ProjectNameMatchingIntegrationTest.groovy

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.execution.taskpath
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class ProjectNameMatchingIntegrationTest extends AbstractIntegrationSpec {
+
+    def setup() {
+        settingsFile << """
+            rootProject.name = "root"
+
+            include("projectA")
+            include("projectA:projectSubA")
+            includeBuild("includedBuild")
+        """.stripIndent()
+
+        // Set-up subprojects
+        file("projectA/projectSubA").mkdirs()
+
+        // Set-up included project
+        file("includedBuild/projectI/projectSubI").mkdirs()
+        file("includedBuild/settings.gradle") << """
+            include("projectI")
+            include("projectI:projectSubI")
+        """.stripIndent()
+    }
+
+    def "logs info message for exact match with subprojects"() {
+        when:
+        run("$projectPath:help", "--info")
+
+        then:
+        outputContains("Found exact project with name '$expectedPath'")
+        outputDoesNotContain("abbrevation")
+
+        where:
+        projectPath | expectedPath
+        // Global root
+        ""  | ":"
+        ":" | ":"
+        // Subprojects
+        ":projectA"             | ":projectA"
+        ":projectA:projectSubA" | ":projectA:projectSubA"
+        // Included build projects
+        ":includedBuild"                      | ":"
+        ":includedBuild:projectI"             | ":projectI"
+        ":includedBuild:projectI:projectSubI" | ":projectI:projectSubI"
+    }
+
+    def "logs info message for pattern match with subprojects"() {
```

Comment:
The test contains coverage not just for subprojects, but also for the root project and for included builds.

```suggestion
    def "logs info message for project name pattern match"() {
```